tests/basis_test.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+
+import numpy
+from scipy import special
+
+from smolyay.basis import BasisFunction, ChebyshevFirstKind, BasisFunctionSet, NestedBasisFunctionSet
+
+@pytest.fixture
+def expected_points_3_set():
+    """extrema for exactness = 8"""
+    return [0, -1.0, 1.0, -1/numpy.sqrt(2), 1/numpy.sqrt(2),
+            -numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75),
+            -numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75),
+            numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75),
+            numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75)]
+
+def test_cheb_initial_zero():
+    """test degree of zero"""
+    f = ChebyshevFirstKind(0)
+    assert f.n == 0
+
+def test_cheb_initial_1():
+    """test initial when degree is 1"""
+    f = ChebyshevFirstKind(1)
+    assert f.n == 1
+    assert f.points == [-1, 1]
+
+def test_cheb_initial_2():
+    """test initial when degree is 2"""
+    f = ChebyshevFirstKind(2)
+    assert f.n == 2
+    assert numpy.allclose(f.points,[-1, 0, 1],atol=1e-10)
+
+def test_cheb_initial_4():
+    """test initial when degree is 4"""
+    expected_points = [-1, -1/numpy.sqrt(2), 0, 1/numpy.sqrt(2), 1]
+    f = ChebyshevFirstKind(4)
+    assert f.n == 4
+    assert numpy.allclose(f.points,expected_points,atol=1e-10)
+
+def test_cheb_initial_8():
+    """test initial when degree is 8"""
+    expected_points = [-1, -numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75), 
+            -1/numpy.sqrt(2), -numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75), 0,
+            numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75), 1/numpy.sqrt(2),
+            numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75), 1]
+    f = ChebyshevFirstKind(8)
+    assert f.n == 8
+    assert numpy.allclose(f.points,expected_points,atol=1e-10)
+
+def test_cheb_call_degree_0():
+    """Chebyshev polynomial degree 0 is 1 for all inputs"""
+    f = ChebyshevFirstKind(0)
+    for i in [-1, -0.5, 0, 0.5, 1]:
+        assert f(i) == 1
+
+def test_cheb_call_degree_1():
+    """Chebyshev polynomial degree 1 should return input"""
+    f = ChebyshevFirstKind(1)
+    for i in [-1, -0.5, 0, 0.5, 1]:
+        assert f(i) == i
```

Comment:
I would fuse these with the tests above, since they are just one more assert after checking the points